services/mqtt/config.go

```diff
@@ -29,8 +29,15 @@ type Config struct {
 	Username string `toml:"username" override:"username"`
 	Password string `toml:"password" override:"password,redact"`
 
-	// NewClientF is a function that returns a client for a given config.
-	NewClientF func(c Config) (Client, error) `toml:"-" override:"-"`
+	// newClientF is a function that returns a client for a given config.
+	// It is used exclusively for testing.
+	newClientF func(c Config) (Client, error) `override:"-"`
+}
+
+// SetNewClientF sets the newClientF on a Config.
+// It is used exclusively for testing.
+func SetNewClientF(cfg *Config, fn func(c Config) (Client, error)) {
```

Comment:
Why not make this a method of the `Config` type? Seems like it would be easier to find that way.